src/more_ops.rs

```diff
@@ -469,17 +469,14 @@ pub fn op_substr<T: Allocator>(a: &mut T, input: T::Ptr) -> Response<T::Ptr> {
     let a0 = args.first()?;
     let s0 = atom(&a0, "substr")?;
     let (n1, _, n2, _) = two_ints(&args.rest()?, "substr")?;
-    let i1: isize = isize::try_from(n1).unwrap_or(isize::max_value());
-    let i2: isize = isize::try_from(n2).unwrap_or(0);
-    let size = s0.len() as isize;
-    if i2 > size || i2 < i1 || i2 < 0 || i1 < 0 {
+    let i1 = u32::try_from(n1).unwrap_or(u32::max_value());
+    let i2 = u32::try_from(n2).unwrap_or(0);
```

Comment:
This seems like it would pin negative numbers to `0`, which may not be what we want.

Reply:
(I realize this isn't new code)

Reply:
ah, yes. negative numbers should be failures. We have tests for that though, right? I'll look into this

Reply:
we have this test:
```
clvm/tests/brun/substr-3.txt
brun -c '(substr (q "abcdefghijkl") 2 5)' '(-1 12)'
FAIL: invalid indices for substr ("abcdefghijkl" -1 12)
```

Reply:
this is a real issue that isn't covered by any test case. I'll fix it and add a test case (to the `clvm` repo)